x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/action/TransportGetTopNFunctionsAction.java

```diff
@@ -116,8 +127,10 @@ static GetTopNFunctionsResponse buildTopNFunctions(GetStackTracesResponse respon
                     }
                 });
             }
+        });
+        if (lostStackTraces.get() != 0) {
+            log.warn("Lost {} stacktraces.", lostStackTraces);
```

Comment:
Similarly to flamegraphs we should rework this. At the very least this should only be a debug log statement.